src/Sylius/Component/Core/Cart/Context/ShopBasedCartContext.php

```diff
@@ -84,6 +99,21 @@ private function setCustomerAndAddressOnCart(OrderInterface $cart, CustomerInter
         }
     }
 
+    private function resolveByGuestFlag(): bool
```

Comment:
I would definitely put this whole logic into a separate service, as this way it's not customisable at all. btw, private function is placed above public 🐎